scripts/nn/examples/AttentionExample.dml

```diff
@@ -0,0 +1,442 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+# We implement a simple example using the basic self-attention
+# mechanism in combination with a LSTM recurrent layer.
+#
+# We use the clickbait dataset
+# (https://www.kaggle.com/datasets/amananandrai/clickbait-dataset?select=clickbait_data.csv)
+# which is a simple binary text classification with 32000 samples.
+#-------------------------------------------------------------
+
+
+source("nn/layers/attention.dml") as attention
+source("nn/layers/affine.dml") as affine
+source("nn/layers/lstm.dml") as lstm
+source("nn/layers/relu.dml") as relu
+source("nn/layers/sigmoid.dml") as sigmoid
+source("nn/optim/adam.dml") as adam
+source("nn/layers/log_loss.dml") as log_loss
+
+
```

Comment:
Is it necessary to write a script to fetch the data?

Reply:
it would be nice to have yes,
simply write something like `wget https://super.nice.webpage/dataset.csv` 
inside a download_attentionExample.sh script.
and make sure that you add a git ignore for it.

If you cannot make such a link then we can put the data set on our Apache webpage and download from there.

Reply:
It is better to upload files to our website: https://github.com/apache/systemds-website#how-to-add-new-dataset

Then download it from there. Ideally, we are not supposed to keep datasets in the project repo.

Reply:
Please review https://github.com/apache/systemds-website/pull/110.